crypto3/libs/blueprint/include/nil/blueprint/bbf/circuit_builder.hpp

```diff
@@ -0,0 +1,668 @@
+                        bp.define_dynamic_table(name,table_specs);
+                    }
+
+                    // this is where we pack lookup tables into constant columns and assign them selectors
+                    std::size_t usable_rows = std::pow(2, std::ceil(std::log2(rows_amount))) - 1; // assure minimum inflation after padding
+                    const auto &lookup_table_ids = bp.get_reserved_indices();
+                    const auto &lookup_tables = bp.get_reserved_tables();
+                    const auto &dynamic_tables = bp.get_reserved_dynamic_tables();
+
+                    std::vector<std::string> ordered_table_names =
+                        nil::crypto3::zk::snark::get_tables_ordered_by_rows_number<FieldType>(lookup_tables);
+
+                    std::size_t start_row = 1;
+                    std::vector<plonk_lookup_table> bp_lookup_tables(lookup_table_ids.size());
+                    std::size_t start_constant_column = presets.constants_amount();
+                    std::size_t prev_columns_number = 0;
+                    int full_selector_id = 0;
+
+                    for(const auto &table_name : ordered_table_names) {
+                        const auto &table = lookup_tables.at(table_name);
+
+                        if (table->get_rows_number() > usable_rows) {
+                            std::size_t options_number = table->get_rows_number() / usable_rows + 1;
+                            start_constant_column += prev_columns_number;
+                            std::size_t cur_constant_column = start_constant_column;
+                            prev_columns_number = options_number * table->get_columns_number();
+
+                            if (presets.constants_amount() < start_constant_column + prev_columns_number) {
+                                presets.resize_constants(start_constant_column + prev_columns_number);
+                            }
+                            // assure all added columns have same amount of usable_rows and need no resizement later
+                            for(std::size_t i = start_constant_column; i < start_constant_column + prev_columns_number; i++)
+                                presets.constant(i,usable_rows-1) = 0;
+
+                            if (full_selector_id == 0) {
+                                row_selector selector_column(usable_rows);
+                                for(std::size_t i = 1; i < usable_rows; i++)
+                                    selector_column.set_row(i);
+                                full_selector_id = gates.add_selector(selector_column);
+                            }
+
+                            if (table->get_rows_number() % usable_rows == 0) options_number--;
+                            std::size_t cur = 0;
+                            for(std::size_t i = 0; i < options_number; i++) {
+                                for(std::size_t start_row = 1; start_row < usable_rows; start_row++, cur++) {
```

Comment:
You already have a variable named start_row, please don't reuse the name.

Reply:
fixed